app/com/linkedin/drelephant/mapreduce/heuristics/DistributedCacheLimitHeuristic.java

```diff
@@ -0,0 +1,148 @@
+        logger
+            .warn("Error parsing " + DISTRIBUTED_CACHE_FILE_SIZE_LIMIT_CONF + " from the conf file. Check for typos...",
+                e);
+      }
+    }
+  }
+
+  @Override
+  public HeuristicResult apply(MapReduceApplicationData data) {
+    if (data == null || !data.getSucceeded()) {
+      return null;
+    }
+    
+    String jobName = data.getJobName();
+    Properties jobConf = data.getConf();
+    String cacheFiles = jobConf.getProperty(MAPREDUCE_JOB_CACHE_FILES, null);
+    String cacheFileSizes = jobConf.getProperty(MAPREDUCE_JOB_CACHE_FILES_FILESIZES, null);
+
+    HeuristicResult result = null;
+
+    if (cacheFiles != null && cacheFileSizes != null) {
+      result =
+          new HeuristicResult(_heuristicConfData.getClassName(), _heuristicConfData.getHeuristicName(), Severity.NONE,
+              0);
+      String[] cacheFilesArray = cacheFiles.split(",");
+      String[] cacheFileSizesArray = cacheFileSizes.split(",");
+
+      if (cacheFilesArray.length != cacheFileSizesArray.length) {
+        result.setSeverity(Severity.MODERATE);
+        logger.warn("Mismatch in the number of files and their corresponding sizes for " + MAPREDUCE_JOB_CACHE_FILES);
+        result.addResultDetail(MAPREDUCE_JOB_CACHE_FILES, Integer.toString(cacheFilesArray.length));
+        result.addResultDetail(MAPREDUCE_JOB_CACHE_FILES_FILESIZES, Integer.toString(cacheFileSizesArray.length));
+        return result;
+      }
+
+      Map<String, String> cacheFileToSizeMap = new HashMap<String, String>();
+      for (int i = 0; i < cacheFilesArray.length; i++) {
+        cacheFileToSizeMap.put(cacheFilesArray[i], cacheFileSizesArray[i]);
+      }
+
+      if (checkFileSizeLimit(result, cacheFileToSizeMap)) {
+        result.setSeverity(Severity.MODERATE);
+        logger.warn("Cache file size was found to be exceeding the max limit of " + distributedCacheFileSizeLimit
+            + " bytes for job " + jobName);
+      }
+    }
+
+    String archiveCacheFiles = jobConf.getProperty(MAPREDUCE_JOB_CACHE_ARCHIVES, null);
+    String archiveCacheFileSizes = jobConf.getProperty(MAPREDUCE_JOB_CACHE_ARCHIVES_FILESIZES, null);
+
+    if (archiveCacheFiles != null && archiveCacheFileSizes != null) {
+
+      if (result == null) {
+        result =
+            new HeuristicResult(_heuristicConfData.getClassName(), _heuristicConfData.getHeuristicName(), Severity.NONE,
+                0);
+      }
+
+      String[] archiveCacheFilesArray = archiveCacheFiles.split(",");
+      String[] archiveCacheFileSizesArray = archiveCacheFileSizes.split(",");
+
+      if (archiveCacheFilesArray.length != archiveCacheFileSizesArray.length) {
+        result.setSeverity(Severity.MODERATE);
+        logger
+            .warn("Mismatch in the number of files and their corresponding sizes for " + MAPREDUCE_JOB_CACHE_ARCHIVES);
+        result.addResultDetail(MAPREDUCE_JOB_CACHE_ARCHIVES, Integer.toString(archiveCacheFilesArray.length));
```

Comment:
How big can this get ? I think we have a limit on this field of 255 chars. This can easily exceed that. You should cut it off at that to avoid db update errors.